File: Python/3_sonar.py
# -*- coding: utf-8 -*-
import time
import sys
from serial import Serial

def main():
    if len(sys.argv) != 2:
        print "usage: python %s <serial>" % __file__
        sys.exit(2)

    serial = Serial(sys.argv[1], timeout=5)

    while True:
        serial.write("S")
        distance = serial.readline().strip()
        sys.stdout.write("Distance is %scm     \r" % distance)
        sys.stdout.flush()
        time.sleep(1)

if __name__ == '__main__':
    main()

File: Ruby/3_sonar.rb
#!/usr/bin/env ruby

require 'serialport'

def check_distance(port)
  serial = SerialPort.new("#{port}", 9600, 8, 1)
  serial.read_timeout = 5000

  loop do
    serial.write "S"
    distance = serial.readline.strip
    print "Distance is #{distance} cm \r"
    STDOUT.flush
    sleep(1)
  end
end

if __FILE__ == $0
  if ARGV.empty?
    puts "usage: ruby #{__FILE__} <serial>"
  else
    check_distance(ARGV[0])
  end
end

